tests/components/snapcast/test_init.py

```diff
@@ -0,0 +1,104 @@
+"""Test the Snapcast module."""
+
+import socket
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.snapcast.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from . import create_mock_snapcast
+
+TEST_CONNECTION = {CONF_HOST: "snapserver.test", CONF_PORT: 1705}
+
+
+async def test_success(hass: HomeAssistant) -> None:
+    """Test successful connection."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert not result["errors"]
+
+    mock_connection = create_mock_snapcast()
+
+    with patch(
+        "snapcast.control.create_server",
+        return_value=mock_connection,
+    ) as mock_setup_entry:
```

Comment:
This is not the entry you are mocking, it is the server creation.
```suggestion
    ) as mock_create_server:
```